features/src/main/scala/com/salesforce/op/features/types/Text.scala

```diff
@@ -181,11 +182,11 @@ class URL(value: Option[String]) extends Text(value){
   /**
    * Extracts url domain, i.e. 'salesforce.com', 'data.com' etc.
    */
-  def domain: Option[String] = value map (new java.net.URL(_).getHost)
+  def domain: Option[String] = value map (s => new java.net.URL(new URI(s, false).toString).getHost)
```

Comment:
I would rather make it a parameter with a default value, e.g.
```scala
/**
 * Extracts url domain, i.e. 'salesforce.com', 'data.com' etc.
 *
 * @param escaped true if URI character sequence is in escaped form. false otherwise. 
 */ 
def domain(escaped: Boolean = false): Option[String] = value map (s => new java.net.URL(new URI(s, escaped).toString).getHost)
```